contract/src/test-integration/java/io/neow3j/contract/NeoTokenIntegrationTest.java

```diff
@@ -0,0 +1,173 @@
+import static io.neow3j.contract.IntegrationTestHelper.walletClients12;
+import static io.neow3j.transaction.Signer.calledByEntry;
+import static io.neow3j.utils.Await.waitUntilTransactionIsExecuted;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import io.neow3j.NeoTestContainer;
+import io.neow3j.crypto.ECKeyPair;
+import io.neow3j.protocol.Neow3j;
+import io.neow3j.protocol.http.HttpService;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Map;
+
+public class NeoTokenIntegrationTest {
+
+    private static Neow3j neow3j;
+    private static NeoToken neoToken;
+
+    @ClassRule
+    public static NeoTestContainer neoTestContainer =
+            new NeoTestContainer("/node-config/config.json");
+
+    @BeforeClass
+    public static void setUp() throws Throwable {
+        neow3j = Neow3j.build(new HttpService(getNodeUrl(neoTestContainer)));
+        neoToken = new NeoToken(neow3j);
+        fundAccountsWithGas(neow3j, client1, client2);
+    }
+
+    @Test
+    public void testUnclaimedGas() throws IOException {
+        long blockHeight = neow3j.getBlockCount().send().getBlockIndex().longValue();
+        BigInteger client1UnclaimedGas = neoToken.unclaimedGas(committee, blockHeight);
+        assertThat(client1UnclaimedGas, greaterThanOrEqualTo(new BigInteger("100000000")));
+    }
+
+    @Test
+    public void testRegisterAndUnregisterCandidate() throws Throwable {
+        Map<ECKeyPair.ECPublicKey, Integer> candidates = neoToken.getCandidates();
+        assertThat(candidates.entrySet(), hasSize(0));
+
+        Hash256 txHash = neoToken.registerCandidate(client1.getECKeyPair().getPublicKey())
+                .wallet(walletClients12)
+                .signers(calledByEntry(client1.getScriptHash()))
+                .sign()
+                .send()
+                .getSendRawTransaction()
+                .getHash();
+        waitUntilTransactionIsExecuted(txHash, neow3j);
+
+        candidates = neoToken.getCandidates();
+        assertThat(candidates.entrySet(), hasSize(1));
+        assertThat(candidates.get(client1.getECKeyPair().getPublicKey()), greaterThanOrEqualTo(0));
+
+        txHash = neoToken.unregisterCandidate(client1.getECKeyPair().getPublicKey())
+                .wallet(walletClients12)
+                .signers(calledByEntry(client1.getScriptHash()))
+                .sign()
+                .send()
+                .getSendRawTransaction()
+                .getHash();
+        waitUntilTransactionIsExecuted(txHash, neow3j);
+
+        candidates = neoToken.getCandidates();
+        assertThat(candidates.entrySet(), hasSize(0));
+    }
+
+    @Test
+    public void testRegisterAndVote() throws Throwable {
```

Comment:
This test has no asserts.

Reply:
I fixed this test in the commit b39a19d. Have a look. :relaxed: